### PR TITLE
HDFS-14889. Ability to check if a block has a replica on provided storage.

### PR DESCRIPTION
**Pull request by ashvina**
Addresses https://issues.apache.org/jira/browse/HDFS-14889.
Adding a method in `BlockInfo` to return true of the block has any replica on a Provided storage.

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 39 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1081 | trunk passed |
| +1 | compile | 65 | trunk passed |
| +1 | checkstyle | 46 | trunk passed |
| +1 | mvnsite | 72 | trunk passed |
| +1 | shadedclient | 835 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 84 | trunk passed |
| 0 | spotbugs | 169 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 166 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 63 | the patch passed |
| +1 | compile | 62 | the patch passed |
| +1 | javac | 62 | the patch passed |
| +1 | checkstyle | 39 | the patch passed |
| +1 | mvnsite | 63 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 778 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 82 | the patch passed |
| +1 | findbugs | 173 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 5596 | hadoop-hdfs in the patch failed. |
| +1 | asflicense | 41 | The patch does not generate ASF License warnings. |
| | | 9372 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdfs.server.blockmanagement.TestUnderReplicatedBlocks |
|   | hadoop.hdfs.tools.TestDFSZKFailoverController |
|   | hadoop.hdfs.server.namenode.snapshot.TestRenameWithSnapshots |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1573/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1573 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 04f898c3378b 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 41440ec |
| Default Java | 1.8.0_222 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1573/1/artifact/out/patch-unit-hadoop-hdfs-project_hadoop-hdfs.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1573/1/testReport/ |
| Max. process+thread count | 4004 (vs. ulimit of 5500) |
| modules | C: hadoop-hdfs-project/hadoop-hdfs U: hadoop-hdfs-project/hadoop-hdfs |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1573/1/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

File: hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockInfoContiguous.java [hunk 1]
```diff
@@ -28,12 +29,16 @@
 @InterfaceAudience.Private
 public class BlockInfoContiguous extends BlockInfo {
 
+  private boolean hasProvidedStorage;
+
   public BlockInfoContiguous(short size) {
     super(size);
+    hasProvidedStorage = false;
```
Comment (goiri) [inline hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockInfoContiguous.java]: Can we just set by default as false and not do it here?

File: hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockInfoContiguous.java [hunk 2]
```diff
@@ -77,9 +85,31 @@ boolean removeStorage(DatanodeStorageInfo storage) {
     setStorageInfo(dnIndex, getStorageInfo(lastNode));
     // set the last entry to null
     setStorageInfo(lastNode, null);
+    if (storage.getStorageType() == StorageType.PROVIDED
+        && !hasProvidedStorages()) {
```
Comment (goiri) [inline hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockInfoContiguous.java]: Shouldn't this already guarantee is false?

Comment (goiri) [reply hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockInfoContiguous.java]: I see is the other method which goes over each DN, nevermind.

File: hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockInfoContiguous.java [hunk 3]
```diff
@@ -77,9 +85,31 @@ boolean removeStorage(DatanodeStorageInfo storage) {
...
+    }
     return true;
   }
 
+  @Override
+  boolean isProvided() {
+    return hasProvidedStorage;
+  }
+
+  private boolean hasProvidedStorages() {
```
Comment (goiri) [inline hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockInfoContiguous.java]: High level javadoc.

File: hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockInfoStriped.java [hunk 4]
```diff
@@ -244,6 +244,11 @@ final boolean hasNoStorage() {
     return true;
   }
 
+  @Override
+  boolean isProvided() {
+    return false;
```
Comment (goiri) [inline hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockInfoStriped.java]: Add a comment saying why it's never provided.

File: hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockInfoContiguous.java (hunk 2 above)
Comment (goiri) [reply hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockInfoContiguous.java]: Can you go over the overhead added to the regular pipeline without Provided?

File: hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockInfoContiguous.java (hunk 1 above)
Comment (ashvina) [reply hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockInfoContiguous.java]: I removed it for now.

File: hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockInfoContiguous.java (hunk 2 above)
Comment (ashvina) [reply hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockInfoContiguous.java]: I removed the check and there is no invocation in the regular pipeline now.

File: hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockInfoContiguous.java (hunk 3 above)
Comment (ashvina) [reply hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockInfoContiguous.java]: Added the javadoc.

File: hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockInfoStriped.java (hunk 4 above)
Comment (ashvina) [reply hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockInfoStriped.java]: Added the javadoc.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 50 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1415 | trunk passed |
| +1 | compile | 72 | trunk passed |
| +1 | checkstyle | 53 | trunk passed |
| +1 | mvnsite | 81 | trunk passed |
| +1 | shadedclient | 985 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 82 | trunk passed |
| 0 | spotbugs | 205 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 202 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 73 | the patch passed |
| +1 | compile | 64 | the patch passed |
| +1 | javac | 64 | the patch passed |
| +1 | checkstyle | 43 | the patch passed |
| +1 | mvnsite | 71 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| -1 | shadedclient | 272 | patch has errors when building and testing our client artifacts. |
| +1 | javadoc | 90 | the patch passed |
| -1 | findbugs | 26 | hadoop-hdfs in the patch failed. |
||| _ Other Tests _ |
| -1 | unit | 25 | hadoop-hdfs in the patch failed. |
| 0 | asflicense | 14 | ASF License check generated no output? |
| | | 3669 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1573/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1573 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux ff81f9dd94c3 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 685918e |
| Default Java | 1.8.0_222 |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1573/2/artifact/out/patch-findbugs-hadoop-hdfs-project_hadoop-hdfs.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1573/2/artifact/out/patch-unit-hadoop-hdfs-project_hadoop-hdfs.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1573/2/testReport/ |
| Max. process+thread count | 411 (vs. ulimit of 5500) |
| modules | C: hadoop-hdfs-project/hadoop-hdfs U: hadoop-hdfs-project/hadoop-hdfs |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1573/2/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

File: hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockInfoContiguous.java [hunk 5]
```diff
@@ -80,6 +81,20 @@ boolean removeStorage(DatanodeStorageInfo storage) {
     return true;
   }
 
+  @Override
+  boolean isProvided() {
+    int len = getCapacity();
+    for(int idx = 0; idx < len; idx++) {
```
Comment (goiri) [inline hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockInfoContiguous.java]: space after the for

File: hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockInfoContiguous.java [hunk 6]
```diff
@@ -80,6 +81,20 @@ boolean removeStorage(DatanodeStorageInfo storage) {
...
+      DatanodeStorageInfo cur = getStorageInfo(idx);
+      if(cur != null) {
```
Comment (goiri) [inline hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockInfoContiguous.java]: Space after the if

Comment (ashvina) [reply hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockInfoContiguous.java]: Fixed

File: hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockInfoContiguous.java (hunk 5 above)
Comment (ashvina) [reply hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockInfoContiguous.java]: Fixed

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 41 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1083 | trunk passed |
| +1 | compile | 68 | trunk passed |
| +1 | checkstyle | 49 | trunk passed |
| +1 | mvnsite | 74 | trunk passed |
| +1 | shadedclient | 925 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 92 | trunk passed |
| 0 | spotbugs | 168 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 166 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 64 | the patch passed |
| +1 | compile | 61 | the patch passed |
| +1 | javac | 61 | the patch passed |
| +1 | checkstyle | 40 | the patch passed |
| +1 | mvnsite | 64 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 807 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 79 | the patch passed |
| +1 | findbugs | 169 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 5347 | hadoop-hdfs in the patch failed. |
| +1 | asflicense | 43 | The patch does not generate ASF License warnings. |
| | | 9239 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdfs.server.namenode.TestRedudantBlocks |
|   | hadoop.hdfs.tools.TestDFSZKFailoverController |
|   | hadoop.hdfs.server.namenode.ha.TestStandbyCheckpoints |
|   | hadoop.hdfs.server.namenode.snapshot.TestRenameWithSnapshots |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1573/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1573 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 3924a8d6fc07 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 4c24f24 |
| Default Java | 1.8.0_222 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1573/3/artifact/out/patch-unit-hadoop-hdfs-project_hadoop-hdfs.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1573/3/testReport/ |
| Max. process+thread count | 4100 (vs. ulimit of 5500) |
| modules | C: hadoop-hdfs-project/hadoop-hdfs U: hadoop-hdfs-project/hadoop-hdfs |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1573/3/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

File: hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/blockmanagement/TestBlockInfo.java [hunk 7]
```diff
@@ -64,6 +67,28 @@ public void testAddStorage() throws Exception {
...
+    BlockInfo blockInfo = new BlockInfoContiguous((short) 3);
+
+    DatanodeStorageInfo storage = mock(DatanodeStorageInfo.class);
+    when(storage.getStorageType()).thenReturn(StorageType.PROVIDED);
+    boolean added = blockInfo.addStorage(storage, blockInfo);
+
+    Assert.assertTrue(added);
+    Assert.assertEquals(storage, blockInfo.getStorageInfo(0));
+    Assert.assertTrue(blockInfo.isProvided());
+
```
Comment (goiri) [inline hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/blockmanagement/TestBlockInfo.java]: What about splitting this test into two with a name that describes what we are testing in each?
Another option would be to just have a different blockInfo for the second chunk.

Comment (ashvina) [reply hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/blockmanagement/TestBlockInfo.java]: There exists a test for verifying behavior of add storage, `testAddStorage `. I had originally modified the original test and added the `isProvided` check. It seemed a bit confusing and out of place to me. So I added this test to clearly test both the scenarios. I would prefer this over two split methods. 
I can rename the second blockInfo if that adds value.

Comment (goiri) [reply hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/blockmanagement/TestBlockInfo.java]: I'm more concerned about readability. Renaming the second one is fine with me.

Comment (ashvina) [reply hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/blockmanagement/TestBlockInfo.java]: Thanks !
I renamed it and added one more check.

Comment (ashvina) [reply hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/blockmanagement/TestBlockInfo.java]: The test now appears splittable. Give me a sec and I will improve it.

Comment (ashvina) [reply hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/blockmanagement/TestBlockInfo.java]: Done. Please take a look.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 426 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1215 | trunk passed |
| +1 | compile | 63 | trunk passed |
| +1 | checkstyle | 45 | trunk passed |
| +1 | mvnsite | 70 | trunk passed |
| +1 | shadedclient | 912 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 79 | trunk passed |
| 0 | spotbugs | 175 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 173 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 66 | the patch passed |
| +1 | compile | 57 | the patch passed |
| +1 | javac | 57 | the patch passed |
| +1 | checkstyle | 40 | the patch passed |
| +1 | mvnsite | 65 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 840 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 76 | the patch passed |
| +1 | findbugs | 176 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 5951 | hadoop-hdfs in the patch failed. |
| +1 | asflicense | 35 | The patch does not generate ASF License warnings. |
| | | 10364 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdfs.server.namenode.TestNameNodeMXBean |
|   | hadoop.hdfs.server.namenode.snapshot.TestRenameWithSnapshots |
|   | hadoop.hdfs.TestRollingUpgrade |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1573/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1573 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux dedb0fa0bade 4.15.0-54-generic #58-Ubuntu SMP Mon Jun 24 10:55:24 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / a3fe404 |
| Default Java | 1.8.0_222 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1573/4/artifact/out/patch-unit-hadoop-hdfs-project_hadoop-hdfs.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1573/4/testReport/ |
| Max. process+thread count | 3070 (vs. ulimit of 5500) |
| modules | C: hadoop-hdfs-project/hadoop-hdfs U: hadoop-hdfs-project/hadoop-hdfs |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1573/4/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 365 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1456 | trunk passed |
| +1 | compile | 77 | trunk passed |
| +1 | checkstyle | 57 | trunk passed |
| +1 | mvnsite | 85 | trunk passed |
| +1 | shadedclient | 1007 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 95 | trunk passed |
| 0 | spotbugs | 213 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 210 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 78 | the patch passed |
| +1 | compile | 68 | the patch passed |
| +1 | javac | 67 | the patch passed |
| +1 | checkstyle | 47 | the patch passed |
| +1 | mvnsite | 65 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 782 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 82 | the patch passed |
| +1 | findbugs | 195 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 5219 | hadoop-hdfs in the patch failed. |
| +1 | asflicense | 42 | The patch does not generate ASF License warnings. |
| | | 10008 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdfs.server.namenode.TestRedudantBlocks |
|   | hadoop.hdfs.server.namenode.ha.TestPendingCorruptDnMessages |
|   | hadoop.hdfs.server.namenode.snapshot.TestRenameWithSnapshots |
|   | hadoop.hdfs.TestReconstructStripedFile |
|   | hadoop.hdfs.server.blockmanagement.TestUnderReplicatedBlocks |
|   | hadoop.hdfs.TestDFSInotifyEventInputStreamKerberized |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1573/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1573 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux b3f859c79d77 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / a3fe404 |
| Default Java | 1.8.0_222 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1573/5/artifact/out/patch-unit-hadoop-hdfs-project_hadoop-hdfs.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1573/5/testReport/ |
| Max. process+thread count | 4824 (vs. ulimit of 5500) |
| modules | C: hadoop-hdfs-project/hadoop-hdfs U: hadoop-hdfs-project/hadoop-hdfs |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1573/5/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

File: hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/blockmanagement/TestBlockInfo.java (hunk 7 above)
Comment (goiri) [reply hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/blockmanagement/TestBlockInfo.java]: LGTM

**Comment by virajith** [comment]
Thanks for posting this @ashvina and the review @goiri . Merged this into trunk.

**Pull request closed without merging**